apollo-client/src/main/java/com/ctrip/framework/apollo/spring/boot/ApolloApplicationContextInitializer.java

```diff
@@ -91,7 +92,8 @@ public void initialize(ConfigurableApplicationContext context) {
   protected void initialize(ConfigurableEnvironment environment) {
 
     if (environment.getPropertySources().contains(PropertySourcesConstants.APOLLO_BOOTSTRAP_PROPERTY_SOURCE_NAME)) {
-      //already initialized
+      //already initialized ,  replay the logs that were printed before the logging system was initialized
+      DeferredLogUtil.replayTo();
```

Comment:
In a non-spring-boot environment, this code won't be executed, which means these logs remains in the memory forever?

I think we could do it in a more precise way, e.g.

**1. We only enable DeferredLogger when eager load is enabled**
Since `com.ctrip.framework.apollo.spring.boot.ApolloApplicationContextInitializer#postProcessEnvironment` will be the first lines of code executed, we could set a flag here

**2. In our DeferredLogger implementation, we cache the logs if the flag is true**

**3. We need to find a point to reset the flag to replay the cached logs and stop caching logs**
I assume there would be some sort of events published when the spring logging system is ready but we need to find it out.

Reply:
- The log will not always be stored in the memory. In an environment where the replay condition cannot be triggered, the log will be recycled after one minute

- In the current client environment, I feel that whether it is the current design or the new delayed log design, it is possible to directly use the apollo.bootstrap.enabled tag to replay. The timing of the replay at this time is also more appropriate